build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java

```diff
@@ -30,47 +31,79 @@
     }
 
     public HttpRequest() {
-        requestBuilder = new Request.Builder();
-        requestCompleted = false;
+        this.requestBuilder = new Request.Builder();
     }
 
-    public static HttpRequest createRequest() {
-        return new HttpRequest();
+    public void setMethod(String method) {
+        this.method = method;
     }
 
-    public void setUrl(String url) {
-        this.requestBuilder = this.requestBuilder.url(url);
+    public boolean setUrl(String url) {
+        try
+        {
+            this.requestBuilder = this.requestBuilder.url(url);
+        }
+        catch (Exception e)
+        {
+            return false;
+        }
+        return true;
     }
 
     public void setHeader(String name, String value) {
-        this.requestBuilder = requestBuilder.addHeader(name, value);
+        try
+        {
+            if(name.equalsIgnoreCase("Content-Type"))
+            {
+                this.contentType = MediaType.parse(value);
+            }
+            else
+            {
+                this.requestBuilder = requestBuilder.addHeader(name, value);
+            }
+        }
+        catch (Exception e)
+        {
+
+        }
     }
 
     public void setBody(byte[] body) {
-        this.requestBuilder = this.requestBuilder.method("POST", RequestBody.create(JSON, body));
+        try
+        {
+            this.requestBuilder = this.requestBuilder.method(this.method, RequestBody.create(this.contentType, body));
+        }
+        catch (Exception e)
+        {
+
+        }
     }
 
     public boolean isRequestCompleted() {
-        return requestCompleted;
+        return this.requestCompleted;
     }
 
-    public void sendRequest() {
-        this.requestCompleted = false;
-        httpClient.newCall(this.requestBuilder.build()).enqueue(new Callback() {
-            @Override
-            public void onFailure(final Call call, IOException e) {
-                onRequestFailed(e.getClass().getCanonicalName());
-            }
-
-            @Override
-            public void onResponse(Call call, final Response response) throws IOException {
-                //if (!response.isSuccessful()) {
-                //    throw new IOException("Unexpected code " + response);
-                //} else {
+    public boolean sendRequest() {
+        try
+        {

```

Comment:
We already have exceptions inside of each call. Do we need this try { catch { block?

Reply:
This calling throws exception when the fail create request.